api/src/test/java/com/messagebird/ContactTest.java

```diff
@@ -62,12 +65,33 @@ public static void tearDown() throws UnauthorizedException, GeneralException, No
 
     @Test
     public void testList() throws UnauthorizedException, GeneralException {
-        ContactList actual = messageBirdClient.listContacts();
+        final ContactList contactResponseList = TestUtil.createContactList();
+
+        MessageBirdService messageBirdServiceMock = mock(MessageBirdService.class);
+        MessageBirdClient messageBirdClientInjectMock = new MessageBirdClient(messageBirdServiceMock);
+
+        when(messageBirdServiceMock.requestList(Mockito.eq(CONTACTPATH), Mockito.eq(0), Mockito.eq(20),
+                Mockito.eq(ContactList.class)))
+                .thenReturn(contactResponseList);
+
+        final ContactList response = messageBirdClientInjectMock.listContacts();
+        verify(messageBirdServiceMock, times(1))
+                .requestList(Mockito.eq(CONTACTPATH), Mockito.eq(0), Mockito.eq(20),
+                        Mockito.eq(ContactList.class));
+        assertEquals(response.getItems().get(0).getFirstName(), contactResponseList.getItems().get(0).getFirstName());
```

Comment:
my java skillz are not amazing but wouldn't it be better to recursively assert that the items are the same? 

with `Assert.assertThat(expected, new ReflectionEquals(actual, excludeFields));` or something? To avoid the repetitive code?

Reply:
When I used ReflectionEquals, I got class cast exception related org.hamcrest.Matcher.
Then, I added 'org.unitils' dependency to the project. We can use "assertReflectionEquals" method which compares all values between two objects. But if you don't like adding a new dependency, I will try to solve hamcrest.Matcher problem.